### ISSUE TITLE
Disabling all but TLS 1.2 breaks MilestonePSTools

### ISSUE THREAD

**Issue by joshooaj** [MilestoneSystemsInc/PowerShellSamples#54]
An internal report from JMT reflects that with encryption enabled between the Management Server and Recording Server, the communication uses TLS 1.2, but MilestonePSTools is seems to be using TLS1. And using the IISCrypto utility to disable all but TLS 1.2 in Windows, MilestonePSTools can no longer connect.

It is _expected_ that MilestonePSTools should negotiate and use the most recent/secure version of TLS, and that there should be no hard-coded dependency on any particular TLS version. This should be more or less dictated by .NET and the underlying OS. It is also expected that if Smart Client and Management Client can connect, MilestonePSTools can connect.

**Comment by joshooaj** [comment]
JMT tested the LogOnServer sample and it worked fine with only TLS 1.2 disabled, so this is definitely a bug unique to MilestonePSTools.

**Comment by joshooaj** [comment]
This should be resolved as MilestonePSTools 23.2.1 now explicitly sets the `[Net.ServicePointManager]::SecurityProtocol` property to include `[Net.SecurityProtocolType]::Tls12`. On certain OS/.NET Framework/PowerShell combinations, the `SystemDefault` protocol value does not include TLS 1.2, so now MilestonePSTools will, on import, update the SecurityProcol value to include any TLS version greater than, or equal to TLS 1.2 with the following code:

```powershell
# Enable the use of any TLS protocol version greater than or equal to TLS 1.2
$protocol = [Net.SecurityProtocolType]::SystemDefault
[enum]::GetNames([Net.SecurityProtocolType]) | Where-Object {
    # Match any TLS version greater than 1.1
            ($_ -match 'Tls(\d)(\d+)?') -and ([version]("$($Matches[1]).$([int]$Matches[2])")) -gt 1.1
} | ForEach-Object { $protocol = $protocol -bor [Net.SecurityProtocolType]::$_ }
[Net.ServicePointManager]::SecurityProtocol = $protocol
```
